### PR TITLE
Development

### PR DESCRIPTION
**Pull request by codeDragon491**

### CODE REVIEW

File: src/components/VonayeProgress.test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import VoyageProgress from './VoyageProgress'
+
+test('render port names', () => {
```
Comment (Tacsiazuma) [inline src/components/VonayeProgress.test.tsx]: You can use the `describe` to categorize the tests.

File: src/components/VonayeProgress.test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        portOfDischarge="Oslo"
+        departureTime="Mon, Jul 8 2022 10:00"
+        arrivalTime="Mon, Jul 8 2022 18:00"/>
+        )
+    const portOfLoading = screen.getByText('Copenhagen')
+    const portOfDischarge = screen.getByText('Oslo')
+    expect(portOfLoading && portOfDischarge).toBeInTheDocument()
+  });
+
+test('render correct number of blue dots at a given progress', () => {
```
Comment (Tacsiazuma) [inline src/components/VonayeProgress.test.tsx]: Is it blue dots or dark dots?

Comment (Tacsiazuma) [reply src/components/VonayeProgress.test.tsx]: The test is like a story. It should not say what it does but why that happens. If I create a roman number converter and have tests like 'should return 333 for CCCXXXIII' and 'should return 87 for LXXXVII' that won't help the reader to figure out whats going on. What is the correct number? I have to check the test in order to figure out that every second progress creates a new dot. I'd state this inside the test name.

File: src/components/VonayeProgress.test.tsx (hunk 1 above)
Comment (Tacsiazuma) [inline src/components/VonayeProgress.test.tsx]: There are two names here. What if the first one is correct and the second one is not. I have to go here and check what went wrong because the test name does not tells me. I'd split this up to loading and discharge.

File: src/components/VoyageProgress.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+interface Times {
+  departureTime: string | Date;
+  arrivalTime: string | Date;
+}
+
+type Props = Ports & Times;
+
+const VoyageProgress: React.FC<Props> = (props) => {
+
+  const departureTime: number = new Date(props.departureTime).getTime()
```
Comment (Tacsiazuma) [inline src/components/VoyageProgress.tsx:61]: The eslint is disabled here as well.

File: src/components/VonayeProgress.test.tsx (hunk 2 above)
Comment (codeDragon491) [reply src/components/VonayeProgress.test.tsx]: dark blue dots more precisely

File: src/components/VonayeProgress.test.tsx (hunk 1 above)
Comment (codeDragon491) [reply src/components/VonayeProgress.test.tsx]: you mean to split it up to two different tests ?

File: src/components/VonayeProgress.test.tsx (hunk 2 above)
Comment (codeDragon491) [reply src/components/VonayeProgress.test.tsx]: Do you think the test itself is good ? Wouldn´t it be better to have the progress dynamic? Is there a way to import the state variable progress into the test and use it there?

File: src/components/VonayeProgress.test.tsx (hunk 1 above)
Comment (Tacsiazuma) [reply src/components/VonayeProgress.test.tsx]: Yes, so whenever a test fails, you'll know exactly what went wrong instead of "this or.. this". As these are fast, it shouldn't matter if it render once or twice.

File: src/components/VonayeProgress.test.tsx (hunk 2 above)
Comment (Tacsiazuma) [reply src/components/VonayeProgress.test.tsx]: There is a concept of data providers although I am not a big fan of it. The point is to test the edge or corner cases. What if the progress is 0, what if its 10, what if its in the middle?

Comment (codeDragon491) [reply src/components/VonayeProgress.test.tsx]: how does this look? 

```
it('render 10 dark blue dots when progress is 100%', () => {
    const { container } = render(<VoyageProgress   
        portOfLoading="Copenhagen" 
        portOfDischarge="Oslo"
        departureTime="Sat, Oct 1 2022 09:00"
        arrivalTime="Sat, Oct 1 2022 17:00"/>
        )
    const progress = 100 / 10
    const darkDots = container.getElementsByClassName("dark-dot");
    // needs to be divided by 2 because componnet renders twice
    expect(darkDots.length / 2).toBe(progress);
  }); 
```
